db-service/lib/common/DatabaseService.js

```diff
@@ -85,7 +85,8 @@ class DatabaseService extends cds.Service {
       try {
         await this.send('ROLLBACK')
       } finally {
-        this.release()
+        // Destroy connection as it might be tainted by parallel operations
+        this.destroy()
```

Comment:
Should we really destroy the connection on each rollback? 
Only when an error happened in acquire or begin, isn't it?

Reply:
no, this shall prevent executing statements prepared before the rollback.